runtime-sdk/src/modules/core/mod.rs

```diff
@@ -136,6 +136,19 @@ pub enum Error {
     TxSimulationFailed(#[from] TxSimulationFailure),
 }
 
+impl Error {
+    /// Generate a proper OutOfGas error, depending on whether the module is configured to emit gas
+    /// use information or not.
+    pub fn out_of_gas<Cfg: Config>(limit: u64, wanted: u64) -> Self {
+        if Cfg::EMIT_GAS_USED_EVENTS {
```

Comment:
this already existed? what did it do before?

Reply:
It suppresses emission of gas used events (which are by default emitted for every transaction). With this PR the same flag is also used for masking exact amounts in out of gas errors.

Reply:
:+1: